cognite/client/_api/datapoints_subscriptions.py

```diff
@@ -246,9 +245,8 @@ def iterate_data(
             res = self._post(url_path=self._RESOURCE_PATH + "/data/list", json=body)
             batch = _DatapointSubscriptionBatch._load(res.json())
 
-            yield batch.updates, batch.subscription_changes
-            if not batch.has_next:
-                return
+            yield DatapointSubscriptionBatch(batch.updates, batch.subscription_changes, batch.has_next)
+
```

Comment:
Do we need both of these classes `_DatapointSubscriptionBatch` and `DatapointSubscriptionBatch`?

Reply:
I guess it makes sense to have both as we do not want to expose the `partitions` to the end user. However, I suggest making these into a hierarchy: `_DatapointSubscriptionBatchInternal(DatapointSubscriptionBatch)`.